mitodl/ocw-studio
# Allow for null checksums returned by Google Drive, ignore files without download links

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #648 

#### What's this PR do?
Allows `DriveFile.checksum` to be null because that's what Google Drive returns for some files.
Ignores any files that have null download links (because that link is used to upload to S3).
Also surrounds the body of `gdrive_sync.api.process_file_result` to log any errors instead of raising them.

#### How should this be manually tested?
- Set DRIVE_*, AWS_* env variables to the same as on RC
- For a new or existing site, create a new google doc or sheet in the `files_final` subfolder beneath your site's folder in[ google drive](https://drive.google.com/drive/folders/1H4HCvbmY7v5YZFeqSlbCI1TFC5MXTMY4).  Also upload some other files there.
- Click the `Sync w/Google Drive' button on your site's resources page.  Wait a minute or so, then refresh the page.  Your uploaded files should appear but not the google doc/sheet.


#### Any background context you want to provide?
Since I'm not 100% positive that null checksums occur only for native google drive files without download links, I changed the code to allow those through as long as they have that download link.  But S3 uploads require the download link, so those files will be ignored.  Support for native google docs/sheets will be added later (#668).
